internal/supervisor/installation.go

```diff
@@ -1456,6 +1489,46 @@ func getAnnotationsNames(annotations []*model.Annotation) []string {
 	return names
 }
 
+func (s *InstallationSupervisor) manageInstallationResourcesCache() {
+	if !s.scheduling.balanceInstallations {
+		return
+	}
+
+	s.logger.Info("Starting installation supervisor cache manager")
+
+	// Create new logger with error level logging to keep logs clean.
+	cacheLogger := log.New().WithFields(log.Fields{
+		"instance": s.instanceID,
+		"cache":    "cluster-resources",
+	})
+	cacheLogger.Logger.SetLevel(log.ErrorLevel)
+
+	for {
+		clusters, err := s.store.GetClusters(&model.ClusterFilter{Paging: model.AllPagesNotDeleted()})
+		if err != nil {
+			cacheLogger.WithError(err).Warn("Failed to query clusters")
+		} else {
+			for _, cluster := range clusters {
+				clusterResources, err := s.provisioner.GetClusterResources(cluster, true, cacheLogger)
+				if err != nil {
+					cacheLogger.WithError(err).Error("Failed to get cluster resources")
+					continue
```

Comment:
Should we clean up the cache entry here? If something breaks in `GetClusterResources` we risk using outdated data for a potentially long time.